core/src/main/java/com/linecorp/armeria/client/Endpoint.java

```diff
@@ -433,6 +438,96 @@ public String authority() {
         return this.authority = authority;
     }
 
+    /**
+     * Converts this endpoint into a URI using the {@code scheme}.
+     *
+     * @param scheme the {@code scheme} for {@link URI}.
+     *
+     * @return the URI
+     */
+    public URI toUri(String scheme) {
+        requireNonNull(scheme, "scheme");
+
+        return toUri(Scheme.parse(scheme));
+    }
+
+    /**
+     * Converts this endpoint into a URI using the {@code scheme} and {@code path}.
+     *
+     * @param scheme the {@code scheme} for {@link URI}.
+     * @param path the {@code path} for {@link URI}.
+     *
+     * @return the URI
+     */
+    public URI toUri(String scheme, String path) {
+        requireNonNull(scheme, "scheme");
+        requireNonNull(path, "path");
+
+        return toUri(Scheme.parse(scheme), path);
```

Comment:
This will fail if a user specified `"http"` rather than `"none+http"`. Is this an intended behavior? How about calling `new URI(scheme, authority(), path, ...)` directly here?